src/components/data/PathListAutomation.js

```diff
@@ -0,0 +1,412 @@
+                const err_code = getErrorCode(error);
+                switch (err_code) {
+                    case ERROR_CODES.ERR_EXISTS:
+                        errMsg = t("pathListExistError");
+                        onError(err_code);
+                        break;
+
+                    case ERROR_CODES.ERR_NOT_FOUND:
+                        errMsg = t("pathListNoMatchError");
+                        onError();
+                        break;
+
+                    default:
+                        errMsg = t("pathListCreateError");
+                        onError();
+                }
+                setCreatePathListError(error);
+                setErrorMsg(errMsg);
+            },
+        }
+    );
+
+    useEffect(() => {
+        if (!infoTypesCache || infoTypesCache.length === 0) {
+            setInfoTypesQueryEnabled(true);
+        } else {
+            if (infoTypes === null || infoTypes.length === 0) {
+                setInfoTypes(infoTypesCache.types);
+            }
+        }
+    }, [infoTypes, infoTypesCache]);
+
+    const handleToggle = (value) => () => {
+        const currentIndex = selectedInfoTypes.indexOf(value);
+        const newChecked = [...selectedInfoTypes];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setSelectedInfoTypes(newChecked);
+    };
+
+    const handlePathListCreation = (values, actions) => {
+        const { selectedPaths, pattern, dest, fileName, foldersOnly } = values;
+
+        const submission = {
+            paths: selectedPaths,
+            dest: `${dest}/${fileName}`,
+            pattern,
+            foldersOnly: foldersOnly || false,
+            recursive: true,
+            requestedInfoType,
+            selectedInfoTypes,
+        };
+
+        const onError = (err_code) => {
+            if (err_code === ERROR_CODES.ERR_EXISTS) {
+                actions.setFieldError("dest", t("pathListExistError"));
+            }
+        };
+        createPathListFile({
+            submission,
+            onError,
+        });
+    };
+
+    const validate = (values) => {
+        const { selectedPaths, dest, fileName } = values;
+        const errors = {};
+        if (!selectedPaths || selectedPaths.length === 0) {
+            errors.selectedPaths = i18nCommon("required");
+        }
+
+        if (!fileName) {
+            errors.fileName = i18nCommon("required");
+        } else {
+            const invalidName = validateDiskResourceName(fileName, t);
+            if (invalidName) {
+                errors.fileName = invalidName;
+            }
+        }
+
+        if (!dest) {
+            errors.dest = i18nCommon("required");
+        }
+        return errors;
+    };
+
+    return (
+        <Formik
+            onSubmit={handlePathListCreation}
+            validate={validate}
+            initialValues={{ selectedPaths: [], dest: startingPath }}
```

Comment:
While testing locally, I got a warning in the console when entering a value for the `pattern` field: `A component is changing an uncontrolled input of type text to be controlled`.

I think this is because there's no initial value set for `pattern` in this `initialValues` object.

According to the Formik docs, every field should have an initial value.
Based on what's getting used in `handlePathListCreation`, there should also be initial values for `fileName`, `pattern`, and `foldersOnly`.
```suggestion
            initialValues={{ selectedPaths: [], dest: startingPath, fileName: "", pattern: "", foldersOnly: false }}
```

Reply:
GTK. I always had my doubts about initialValues.